frame/elections-phragmen/src/lib.rs

```diff
@@ -325,12 +327,68 @@ decl_module! {
 		fn deposit_event() = default;
 
 		const CandidacyBond: BalanceOf<T> = T::CandidacyBond::get();
-		const VotingBond: BalanceOf<T> = T::VotingBond::get();
+		const LegacyVotingBond: BalanceOf<T> = T::LegacyVotingBond::get();
+		const VotingBondBase: BalanceOf<T> = T::VotingBondBase::get();
+		const VotingBondFactor: BalanceOf<T> = T::VotingBondFactor::get();
 		const DesiredMembers: u32 = T::DesiredMembers::get();
 		const DesiredRunnersUp: u32 = T::DesiredRunnersUp::get();
 		const TermDuration: T::BlockNumber = T::TermDuration::get();
 		const ModuleId: LockIdentifier  = T::ModuleId::get();
 
+		fn on_runtime_upgrade() -> Weight {
+			if Self::needs_migrations() {
+				let mut consumed_weight = 0;
```

Comment:
I think it is fine to just take a full block for a runtime upgrade operation.

Reply:
Can you also move this into its own function which is then called by `on_runtime_upgrade`?

This may allow us to keep the code around a little longer and allow other users to trigger the upgrade on their own time.

Reply:
thought so too, idk, it looked easy enough to do it accurately as well. Either way it is fine since it is a one time thing.

Reply:
Looking at it a bit more carefully, you can immediately deprecate `LagacyVotingBond`.

Then create a function:

```
pub fn migrate_deposit<T>(legacy_voting_bond: BalanceOf<T>) {
    ...
}
```

Then call this function from the runtime (not the pallet): https://github.com/paritytech/substrate/pull/7040

And pass the old voting bond in that way. This would avoid yet another breaking change for the pallet users.

Reply:
> Then call this function from the runtime (not the pallet): #7040

Can you explain this a bit more?

Reply:
We can trigger some runtime upgrade logic from the runtime `lib.rs` rather than from the pallet. See here:

https://github.com/paritytech/polkadot/pull/1636

Reply:
Sweet.